MediumProject/src/main/java/stage3/Menu.java

```diff
@@ -0,0 +1,50 @@
+package stage3;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.function.Consumer;
+
+
+public class Menu {
+ private static final Scanner scan = new Scanner(System.in);
+
+    private static String printMenu(){
+        System.out.println("""
+                    === Menu ===
+                    1. Find a person
+                    2. Print all people
+                    0. Exit""");
+        return scan.nextLine();
+
+    }
+    private static final Map<String , Consumer<String>> actionsMap;
+    static {
+        Consumer<String> find = (search) -> {
```

Comment:
variables assignment is redundant + use util map methods

        Consumer<String> find = (search) -> {
            String input = scan.nextLine();
            System.out.println("\nEnter a name or email to search all suitable people.");
           SearchingEngine.searchPerson(input);
        };
        Consumer<String> print = (search) -> {
            UsersDataOutput.printAllPeople();
        };
        actionsMap = new HashMap<>();
        actionsMap.put("1" , find);
        actionsMap.put("2" , print);

=>>>

        Map.of(
"1" , 
(search) -> {
            String input = scan.nextLine();
            System.out.println("\nEnter a name or email to search all suitable people.");
           SearchingEngine.searchPerson(input)),
 "2" ,
 (search) -> {
            UsersDataOutput.printAllPeople());

Reply:
done